### PR TITLE
Find the external meshes in the local file system

### PR DESCRIPTION
**Pull request by S-Dafarra**
cc @GiulioRomualdi 

Fix https://github.com/robotology/idyntree/issues/291 .

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 - Add the possibility to plot and update frames in the Matlab visualizer.
+- Added ``getFileLocationOnLocalFileSystem`` method in ``ExternalMesh`` that attempt to find the mesh location in the local file system
```
Comment (traversaro) [inline CHANGELOG.md:12]: Can we specify that this is now used when using the visualizer?

File: src/model/src/SolidShapes.cpp [hunk 2]
```diff
@@ -194,6 +197,91 @@ namespace iDynTree
...
+            }
+        };
+
+        auto getFilePath = [isFileExisting](const std::string& filename, const std::string& prefixToRemove, const std::unordered_set<std::string>& paths)
+        {
+            if(isFileExisting(filename))
+            {
+                return filename;
+            }
+
```
Comment (traversaro) [inline src/model/src/SolidShapes.cpp:262]: This ensure that the current use case of specifying files without a `package://`  URI file is still supported, right?

**Comment by traversaro** [comment]
cc @singhbal-baljinder this should permit to use the visualizer without any more specific hacks.

**Review by traversaro — CHANGES_REQUESTED** [review]
Just a small comment on the changelog. Thanks!

**Comment by traversaro** [comment]
Related issue: https://github.com/robotology/idyntree/issues/799 (that can be in any case addressed **after** this PR has been merged).

File: CHANGELOG.md (hunk 1 above)
Comment (S-Dafarra) [reply CHANGELOG.md]: Done in https://github.com/robotology/idyntree/pull/798/commits/a98f5ef59f64034542efea28c336d2f7b6be05d5

**Comment by S-Dafarra** [comment]
Actually, there were a couple of bugs in the Windows branching. Indeed, I haven't tested it in WIndows :sweat_smile:

File: src/model/src/SolidShapes.cpp (hunk 2 above)
Comment (S-Dafarra) [reply src/model/src/SolidShapes.cpp:262]: Yes, if ``filename`` is already the path of a file, it simply returns it as it is.

**Comment by traversaro** [comment]
So, did you tested on Windows now? Is this ready to be merged?

**Comment by S-Dafarra** [comment]
> So, did you tested on Windows now? Is this ready to be merged?

I haven't yet. I will

**Comment by S-Dafarra** [comment]
> So, did you tested on Windows now? Is this ready to be merged?

I just tested it in Windows too. It seems to be working (apart from the usual headaches when testing something on Windows)

image

**Review by traversaro — APPROVED** [review]

**Comment by traversaro** [comment]
Perfect, can you resolve the conflicts so that we can merge? Thanks!

**Comment by S-Dafarra** [comment]
Actually, I forgot to mention that it did not work out of the box, simply because right now we don't define the ``GAZEBO_MODEL_PATH`` env variable (neither ``ROS_PACKAGE_PATH`` nor ``AMENT_PREFIX_PATH``).

**Comment by traversaro** [comment]
> 
> 
> Actually, I forgot to mention that it did not work out of the box, simply because right now we don't define the `GAZEBO_MODEL_PATH` env variable (neither `ROS_PACKAGE_PATH` nor `AMENT_PREFIX_PATH`).

I opened an issue for this: https://github.com/robotology/robotology-superbuild/issues/577 .

**Comment by traversaro** [comment]
By the way, how did you installed irrlicht on Windows? vcpkg?

**Comment by S-Dafarra** [comment]
> By the way, how did you installed irrlicht on Windows? vcpkg?

A bit worst than that, just a bit 😁 I downloaded the precompiled version from Sourceforge, I added the ``bin`` folder to the path and set the CMake variables (pointing to the ``include`` folder and the to the ``Irrlicht.lib`` library) manually.

**Pull request merged**